source/common/network/filter_matcher.cc

```diff
@@ -0,0 +1,186 @@
+#include "common/network/filter_matcher.h"
+
+#include "envoy/network/filter.h"
+
+#include "absl/strings/str_format.h"
+
+namespace Envoy {
+namespace Network {
+
+// Introduced to save number of memory allocation at build phase memory usage match phase.
+// Work with OwnedListenerFilterMatcher.
+// The first phase is constructor, requiring only 1 slots.
+// The second phase is complete() which reserves contiguously slots.
```

Comment:
Sorry can't parse this. Can you expand/clarify? Also clarify below what complete() does. It will help me review.

Reply:
Updated: could you take another look?